clustering/web/cache/src/main/java/org/wildfly/clustering/web/cache/session/SessionCreationMetaData.java

```diff
@@ -28,11 +28,31 @@
  * The more static aspects of a session's meta-data.
  * @author Paul Ferraro
  */
-public interface SessionCreationMetaData extends ImmutableSessionCreationMetaData {
-
+public interface SessionCreationMetaData extends ImmutableSessionCreationMetaData, AutoCloseable {
+    /**
+     * Sets the maximum duration of time this session may remain idle before it will be expired by the session manager.
+     * @param a maximum duration of time this session may remain idle before it will be expired by the session manager.
+     */
     void setMaxInactiveInterval(Duration duration);
 
+    /**
+     * Indicates whether or not this session has been invalidated.
+     * @return true, if this session was invalidated, false otherwise.
+     */
     boolean isValid();
 
+    /**
+     * Invalidates this session.
+     * @return true, if this session was previous valid, false otherwise
+     */
     boolean invalidate();
+
+    /**
+     * Indicates whether or not this session was newly created.
```

Comment:
What does 'newly created' mean? First request for the session id anywhere in the cluster? Locally?

Reply:
As defined by: https://javaee.github.io/javaee-spec/javadocs/javax/servlet/http/HttpSession.html#isNew--

Reply:
This property was formerly implied by the fact that the creationTime != the lastAccessedTime.  However, since the semantics of lastAccessedTime have changed, this must now be determined explicitly, hence this logic.

Reply:
Thanks.